modules/redis/redis.c

```diff
@@ -341,7 +341,7 @@ redis_dd_init(LogPipe *s)
   if (!log_threaded_dest_driver_init_method(s))
     return FALSE;
 
-  if (!self->key && !self->param1)
+  if (!g_list_nth_data(self->fields, 0) && !g_list_nth_data(self->fields, 1))
```

Comment:
[optional] you could use `g_list_length(self->fields) >= 2`

Reply:
The old check was buggy, as it allowed to have no `key`, if `param1` was set, but `key` was used regardless in `redis_worker_insert()`, without checking if it existed:

```                                                                                                                                                                                               
  log_template_format(self->key, msg, &self->template_options, LTZ_SEND,                                                                                                                      
                      self->super.worker.instance.seq_num, NULL, self->key_str); 
```
So, I changed it to check if the list is empty.